js-exercises/object-utils/objectUtils.js

```diff
@@ -0,0 +1,59 @@
+// ! code may break didn't consider many edge cases
+
+// ? 1 If Object has Array/function in value
+// ? 2 If user does a arithmetic operation in callBackFunction on string value
+// ? 3 if user want to access/modify - not iteratable for not modifiable key value
+
+// ? Thought of using TRY CATCH block to throw error in case of Error.
+// # Is it a good approach @ ARFAT
+
+
+const errorCheck = (obj) => {
+  if (obj === null) throw new Error('received NULL -- expecting OBJECT ')
+  if (typeof obj !== 'object') throw new Error(`received ${typeof obj} -- expecting OBJECT `);
+  if (Array.isArray(obj)) throw new Error('received ARRAY -- expecting OBJECT ');
+};
+
+const objectToArray = (obj) => Object.entries(obj);
+
+const arrayToObject = (arr) => Object.fromEntries(arr);
+
+// your implementation
+
+const map = (obj, callBackFn) => {
+  errorCheck(obj);
```

Comment:
Check error in the function itself. This way I don't know what the error check does.

Reply:
Changed functino name to ---> throwError()

Reply:
Changed functino name to ---> throwError()